### PR TITLE
Refactor all.sh

### PR DESCRIPTION
**Pull request by mpg**
## Description

Preparation for #9636. Not moving anything to the framework yet, just splitting the files the way we'll want them for moving the shared parts.

## PR checklist

- [x] **changelog** not required because: internal, tests only
- [x] **development PR** here
- [x] **framework PR** not required
- [x] **3.6 PR** provided #9702
- [x] **2.28 PR** not required because: not worth it (also, we already didn't split `components-*.sh` out of `all.sh` there).
- **tests**  not required because: covered by existing tests

### CODE REVIEW

**Review by eleuzi01 — APPROVED** [review]
LGTM

**Review by Harry-Ramsey — APPROVED** [review]
LGTM, thanks.

**Comment by mpg** [comment]
@Harry-Ramsey @eleuzi01 Added two commits while preparing the 3.6 backport, in order to get both branches aligned. Please review again.

File: tests/scripts/all-core.sh [hunk 1]
```diff
@@ -0,0 +1,1023 @@
...
+    if in_mbedtls_repo; then
+        CONFIG_H='include/mbedtls/mbedtls_config.h'
+        if [ -d tf-psa-crypto ]; then
+            CRYPTO_CONFIG_H='tf-psa-crypto/include/psa/crypto_config.h'
+            PSA_CORE_PATH='tf-psa-crypto/core'
+            BUILTIN_SRC_PATH='tf-psa-crypto/drivers/builtin/src'
+        else
+            CRYPTO_CONFIG_H='include/psa/crypto_config.h'
+            # helper_armc6_build_test() relies on these being defined,
+            # but empty if the paths don't exits (as in 3.6).
```
Comment (eleuzi01) [inline tests/scripts/all-core.sh:175]:
```suggestion
            # but empty if the paths don't exist (as in 3.6).
```

**Review by Harry-Ramsey — DISMISSED** [review]
LGTM other than the aforementioned issue.

**Review by Harry-Ramsey — APPROVED** [review]
LGTM

**Pull request merged**